casepro/backend/junebug.py

```diff
@@ -463,3 +467,44 @@ def received_junebug_message(request):
         created_on=datetime.now(), has_labels=True)
 
     return JsonResponse(msg.as_json())
+
+
+@csrf_exempt
+def receive_identity_store_optout(request):
+    '''Handles optout notifications from the Identity Store.'''
+    if request.method != 'POST':
+        return JsonResponse({'reason': 'Method not allowed.'}, status=405)
+
+    try:
+        data = json.loads(request.body)
+    except ValueError as e:
+        return JsonResponse(
+            {'reason': 'JSON decode error', 'details': e.message}, status=400)
+
+    identity_id = data.get('identity_id')
+    optout_type = data.get('optout_type')
+
+    syncer = IdentityStoreContactSyncer()
+
+    # The identity store currently doesn't specify the response format or do
+    # anything with the response.
+
+    # I don't know how to get the org
+    local_contact = syncer.fetch_local(org, identity_id)
```

Comment:
@hodgestar I think you've just pointed out a bug to me :disappointed: . You see, we don't use the `org` when we are getting contacts from the identity store but we do send it along when we sync contacts ([here](https://github.com/praekelt/casepro/blob/EPIC-OPENHELPD-17/casepro/backend/junebug.py#L247-L249)).  Because `org` is used when checking if a contact already exists, this essentially means that if we have multiple `orgs` then there will be a copy of each contact for every `org`